packages/ra-core/src/dataProvider/fetch.spec.ts

```diff
@@ -85,4 +92,40 @@ describe('createHeadersFromOptions', () => {
         );
         expect(headersWithoutMethod.get('Content-Type')).toBeNull();
     });
+
+    it('should not add a Content-Type header if there is no body', () => {
+        const optionsWithDelete = {
+            method: 'DELETE',
+        };
+
+        const headersWithDelete = createHeadersFromOptions(optionsWithDelete);
+        expect(headersWithDelete.get('Content-Type')).toBeNull();
+        const optionsWithDeleteAndBody = {
+            method: 'DELETE',
+            body: JSON.stringify(null),
+        };
+
+        const headersWithDeleteAndBody = createHeadersFromOptions(
+            optionsWithDeleteAndBody
+        );
+        expect(headersWithDeleteAndBody.get('Content-Type')).toStrictEqual(
+            'application/json'
+        );
+    });
+
+    it('should not add a Content-Type header if there already is a Content-Type header', () => {
+        const optionsWithContentType = {
+            headers: new Headers({
```

Comment:
And you should probably use another method than the default, otherwise you're falling back on the test on GET.

Reply:
you're correct!